### PR TITLE
refactor(meta): build actors in inject barrier

### PR DESCRIPTION
**Pull request by wenym1**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

After https://github.com/risingwavelabs/risingwave/pull/18104, when handling `BuildActor` request, we simply spawn the actors and save the join handle. After the process of building actors is simplified, we can further postpone spawning actor till inject barrier, so that
* stream service rpc `broadcast_actor_info_table`, `update_actors` and `build_actors` can be removed, and the works of these rpc can be done together in `InjectBarrier`.
* on CN side, all spawned actors are part of the streaming graph, and we don't need to handle the case of actors that are spawned but not joined in the streaming graph
* we can avoid calling`drop_actors` while cancelling a command, because for a create streaming job command that is enqueued but not handled yet, its actor is not spawned yet, and there is no need to further drop the actors.

In this PR, for simplicity, most changes are on the meta node. The actor locations to be broadcast and the `BuildActorInfo` is included in the `InjectBarrierRequest`. On local barrier manager, we still reuse the current implemented methods. The local barrier manager will be refactored in future PRs.

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added test labels as necessary. See [details](https://github.com/risingwavelabs/risingwave/blob/main/docs/developer-guide.md#ci-labels-guide).
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

**Comment by MrCroxx** [comment]
AMAZING 🥵

File: src/meta/src/barrier/command.rs [hunk 1]
```diff
@@ -844,6 +844,43 @@ impl CommandContext {
         mutation
     }
 
+    pub fn actors_to_create(&self) -> HashMap<WorkerId, Vec<StreamActor>> {
+        match &self.command {
+            Command::CreateStreamingJob { info, job_type } => {
+                let mut map = match job_type {
+                    CreateStreamingJobType::Normal => HashMap::new(),
+                    CreateStreamingJobType::SinkIntoTable(replace_table) => {
+                        replace_table.new_table_fragments.actors_to_create()
```
Comment (yezizp2012) [inline src/meta/src/barrier/command.rs:853]: Great job!! Looks like #14904 can be resolved naturally!

**Review by yezizp2012 — APPROVED** [review]
LGTM!! It definitely simplify the barrier processing and actor building. 🤔 We need to rethink about #16742 and find another solution.

**Review by kwannoel — APPROVED** [review]

File: src/meta/src/barrier/creating_job/status.rs [hunk 2]
```diff
@@ -89,7 +91,12 @@ impl CreatingStreamingJobStatus {
         &mut self,
         is_checkpoint: bool,
     ) -> Option<(
-        Vec<(TracedEpoch, TracedEpoch, BarrierKind)>,
+        Vec<(
```
Comment (BugenZhao) [inline src/meta/src/barrier/creating_job/status.rs:94]: Shall we wrap it into a struct and also update the comments? This doesn't looks that readable.

File: src/meta/src/model/stream.rs [hunk 3]
```diff
@@ -528,13 +528,19 @@ impl TableFragments {
     }
 
     /// Returns actor map: `actor_id` => `StreamActor`.
```
Comment (BugenZhao) [inline src/meta/src/model/stream.rs:530]: May also update the docs.

**Review by BugenZhao — COMMENTED** [review]
My only concern is that this may increase the latency for injecting barriers, which could slow down the barrier manager loop. Rest LGTM.

**Pull request merged**